internal/webhook/v1/alertmanager_config_secret_webhook.go

```diff
@@ -18,80 +18,145 @@ package v1
 	secret, ok := newObj.(*corev1.Secret)
 	if !ok {
 		return nil, fmt.Errorf("expected a Secret object for the newObj but got %T", newObj)
 	}
-	secretlog.Info("Validation for Secret upon update", "name", secret.GetName())
 
-	// TODO(user): fill in your validation logic upon object update.
+	log.Info("Validation for Secret upon update", "name", secret.GetName())
 
-	return nil, nil
+	if err := v.validateTenant(ctx, secret); err != nil {
+		return nil, err
+	}
+	return nil, validateAlertmanagerConfig(ctx, secret)
 }
 
 // ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type Secret.
-func (v *SecretCustomValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
-	secret, ok := obj.(*corev1.Secret)
+func (v *AlertmanagerConfigSecretCustomValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	// We have nothing to validate on deletion
+	return nil, nil
+}
+
+func (v *AlertmanagerConfigSecretCustomValidator) validateTenant(ctx context.Context, secret *corev1.Secret) error {
+	// Check that the secret has the correct labels.
+	tenant, ok := secret.Labels[tenancy.TenantSelectorLabel]
 	if !ok {
-		return nil, fmt.Errorf("expected a Secret object but got %T", obj)
+		return fmt.Errorf("tenant label is required")
 	}
-	secretlog.Info("Validation for Secret upon deletion", "name", secret.GetName())
 
-	// TODO(user): fill in your validation logic upon object deletion.
+	// Check that the tenant is defined in a Grafana Organization.
+	tenants, err := tenancy.ListTenants(ctx, v.client)
+	if err != nil {
+		return fmt.Errorf("failed to list tenants: %w", err)
+	}
+	if !slices.Contains(tenants, tenant) {
+		return fmt.Errorf("tenant %s is not in the list of accepted tenants", tenant)
+	}
 
-	return nil, nil
+	// Check that there is only one alertmanager config for the tenant.
+	var secretList corev1.SecretList
+	err = v.client.List(ctx, &secretList, client.InNamespace(""),
+		client.MatchingLabelsSelector{
+			Selector: labels.SelectorFromSet(
+				labels.Set{
+					predicates.AlertmanagerConfigSelectorLabelName: predicates.AlertmanagerConfigSelectorLabelValue,
+					tenancy.TenantSelectorLabel:                    tenant,
+				},
+			),
+		})
+	if err != nil {
+		return fmt.Errorf("failed to list secrets for tenant %s: %w", tenant, err)
+	}
+
+	if len(secretList.Items) > 0 {
+		err = nil
+		for _, s := range secretList.Items {
+			if s.Name != secret.Name || s.Namespace != secret.Namespace {
+				err = errors.Join(err, fmt.Errorf("secret %s/%s for tenant %s already exists", s.Name, s.Namespace, tenant))
+			}
+		}
+		return err
+	}
+
+	return nil
+}
+
+func validateAlertmanagerConfig(ctx context.Context, secret *corev1.Secret) error {
+	// Check that the secret contains an "alertmanager.yaml" file.
+	alertmanagerConfig, found := secret.Data["alertmanager.yaml"]
```

Comment:
Please re-use the already define value at [`alertmanagerConfigKey`](https://github.com/giantswarm/observability-operator/blob/33a3f15c55fd149538ce0416dbccec9ff7a96ad6/pkg/alertmanager/alertmanager.go#L25C2-L25C23)